harness/determined/cli/user.py

```diff
@@ -155,6 +155,41 @@ def whoami(parsed_args: Namespace) -> None:
     print("You are logged in as user '{}'".format(user.username))
 
 
+@authentication.required
+@login_sdk_client
```

Comment:
I talked to Ryan about the fitness of creating multiple `Authentication` tokens, and he reminded me of [that we don't have to](https://github.com/determined-ai/determined/blob/47b37ebc6bfaaade95f4c095ee4fdf6e7cd10143/harness/determined/common/experimental/determined.py#L65).

All you need is `@authentication.required` and `setup_session`. From there, you can construct your own `Determined` object like

```python
det = determined.common.experimental.determined.Determined._from_session(sess)
det.get_user_by_name(target_user)
```

no extra round-trip to the master to login the SDK client.